lib/manageiq/api/common/open_api/generator.rb

```diff
@@ -176,6 +176,14 @@ def openapi_list_description(klass_name, primary_collection)
                       "schema" => { "$ref" => build_collection_schema(klass_name) }
                     }
                   }
+                },
+                "404" => {
+                  "description" => "#{klass_name.pluralize} collection",
+                  "content"     => {
+                    "application/json" => {
+                      "schema" => { "$ref" => build_collection_schema(klass_name) }
```

Comment:
This renders a 404 error document
```
{
  "errors": [
    {
      "status": 404,
      "detail": "Record not found"
    }
  ]
}
```

Reply:
@bdunne I made this change locally
```diff
diff --git a/lib/manageiq/api/common/open_api/generator.rb b/lib/manageiq/api/common/open_api/generator.rb
index 9efcbda..16528c7 100644
--- a/lib/manageiq/api/common/open_api/generator.rb
+++ b/lib/manageiq/api/common/open_api/generator.rb
@@ -176,16 +176,14 @@ module ManageIQ
                       "schema" => { "$ref" => build_collection_schema(klass_name) }
                     }
                   }
-                },
-                "404" => {
-                  "description" => "#{klass_name.pluralize} collection",
-                  "content"     => {
-                    "application/json" => {
-                      "schema" => { "$ref" => build_collection_schema(klass_name) }
-                    }
-                  },
                 }
-              }
+              },
+              "errors" => [
+                {
+                  "status": 404,
+                  "detail": "Record not found"
+                }
+              ]
```
And that build the 404 like this in the spec -
```
        },
        "errors": [
          {
            "status": 404,
            "detail": "Record not found"
          }
        ]
      },
```
But in other places int he spec we have it this way
```
          },
          "404": {
            "description": "Not found"
          }
        }
      },
```
Which one is correct?

Reply:
The first one is correct, we need the `"errors"` key at the top level of the response.

Reply:
Thanks, @bdunne, I'll work on making the necessary changes